app-backup/luckybackup/patches/luckybackup-0.4.9.patchset

```diff
@@ -0,0 +1,84 @@
+
+
+diff --git a/src/main.cpp b/src/main.cpp
+index 935603e..4f66bba 100644
+--- a/src/main.cpp
++++ b/src/main.cpp
+@@ -33,6 +33,9 @@
+ 
+ #include <QApplication>
+ #include <QTranslator>
++#ifdef Q_OS_HAIKU
++#include <QDir>
++#endif
+ 
+ #include "luckybackupwindow.h"
+ #include "global.h"
+@@ -52,6 +55,10 @@ int main(int argc, char *argv[])
+ {
+     QScopedPointer<QCoreApplication> app(createApplication(argc, argv));
+ 
++#ifdef Q_OS_HAIKU
++    QDir::setCurrent(QCoreApplication::applicationDirPath());
++#endif
++
+     //arguments test
+     if (!argumentsTest(argc,argv))
+         return 1;		// error code1: something wrong with the command arguments - app fail to start (or just --help or --version given as args)
+-- 
+2.19.0
+
+
+From 8ea9eea609a179dfe67852cfab86250d10142f83 Mon Sep 17 00:00:00 2001
+From: fbrosson <fbrosson@localhost>
+Date: Wed, 17 Oct 2018 14:23:21 +0000
+Subject: hardcode paths for Haiku
+
+
+diff --git a/src/global.cpp b/src/global.cpp
+index 1efe3ed..3e0cb0f 100644
+--- a/src/global.cpp
++++ b/src/global.cpp
+@@ -52,8 +52,8 @@ QString appVersionString = "0.4.9";
+ double const validProfileVersion = 0.21;
+ double const validScheduleVersion = 0.34;
+ double const validSettingsVersion = 0.3;
+-QString luckyBackupDir = myHome + "/."+appName+"/";
+-QString luckyBackupDefaultDir = myHome + "/."+appName+"/";
++QString luckyBackupDir = myHome + "/config/settings/LuckyBackup/";
++QString luckyBackupDefaultDir = myHome + "/config/settings/LuckyBackup/";
+ QString settingsFile = luckyBackupDir + "settings.ini";
+ QString profileDir = luckyBackupDir + "profiles/";
+ QString defaultProfile = profileDir + "default.profile";
+@@ -71,18 +71,18 @@ QString snapChangesString = "[changed_data]%i[LB]%n";
+ QString scheduleDir = luckyBackupDir + "schedule/";
+ QString schedulefilename = scheduleDir + "schedule.dat";
+ QString cronfilename = scheduleDir + "luckyCron.txt";
+-QString relativeTransDir = "translations/";
+-QString systemTransDir = "/usr/share/luckybackup/translations/";
++QString relativeTransDir = "../data/luckybackup/translations/";
++QString systemTransDir = "/boot/system/data/luckybackup/translations/";
+ QDir transDir;
+ QTranslator appTranslator;
+ QTranslator translator_qt;
+ QString locale = QLocale::system().name();
+-QString relativeLicense = "license/gpl.html";
+-QString systemLicense = "/usr/share/doc/luckybackup/license/gpl.html";
++QString relativeLicense = "../data/luckybackup/license/gpl.html";
++QString systemLicense = "/boot/system/data/luckybackup/license/gpl.html";
+ QString suseLicense = "/usr/share/doc/packages/luckybackup/license/gpl.html";
+ QString debianLicense = "/usr/share/common-licenses/GPL-3";
+-QString relativeManual = "manual/index.html";
+-QString systemManual = "/usr/share/doc/luckybackup/manual/index.html";
++QString relativeManual = "../data/luckybackup/manual/index.html";
++QString systemManual = "/boot/system/data/luckybackup/manual/index.html";
```

Comment:
@fbrosson 

Are there any reasons why these can't be changed into `@Template@` which then you can `sed` in the recipe?

I'm a bit concerned about the "relative" paths, what exactly are they relative to? As seen last time, they might be relative to the current working dir, which may causes unintended content injection. I'd like to get rid of these relative paths instead.

Reply:
All apps will have to tolerate relative paths eventually, as they can get installed in HOME as well as /system/.

Reply:
Not really, `/package/name-version/.self/` stay the same regardless of installation location. Also, there's `find_path` if you want the fancy `/boot/{home,system}` paths.